code/modules/mob/living/carbon/life.dm

```diff
@@ -384,3 +383,32 @@
 		if(360.15 to INFINITY) //360.15 is 310.15 + 50, the temperature where you start to feel effects.
 			//We totally need a sweat system cause it totally makes sense...~
 			bodytemperature += min((body_temperature_difference / BODYTEMP_AUTORECOVERY_DIVISOR), -BODYTEMP_AUTORECOVERY_MINIMUM)	//We're dealing with negative numbers
+
+/mob/living/carbon/proc/handle_liver()
+	var/obj/item/organ/liver/liver = getorganslot("liver")
+	if(((!(NOLIVER in dna.species.species_traits)) && (!liver)) || liver.damage >= 100)
+		liver.failing = TRUE
+		liver_failure()
+	else
+		liver.failing = FALSE
+
+/mob/living/carbon/proc/undergoing_liver_failure()
+	var/obj/item/organ/liver/liver = getorganslot("liver")
+	if(liver.failing)
+		return TRUE
+
+/mob/living/carbon/proc/return_liver_damage()
+	var/obj/item/organ/liver/liver = getorganslot("liver")
+	if(liver)
+		return liver.damage
+
+/mob/living/carbon/proc/applyLiverDamage(var/obj/item/organ/liver/L, var/d)
+	L.damage += d
+
+/mob/living/carbon/proc/liver_failure()
+	if(reagents.get_reagent_amount("stabiliver"))

```

Comment:
While this is a bit snowflakey, I don't see a better way to handle this, as this will need to run when normal metabolization isn't running.